src/model.js

```diff
@@ -273,11 +288,21 @@ const BookshelfModel = ModelBase.extend({
    *   be the singular form of the `Target` model's tableName, followed by `_id` /
    *   `_{{{@link Model#idAttribute idAttribute}}}`.
    *
+   * @param {string=} foreignKeyTarget
+   *
+   *   Column in this model's table which `foreignKey` references, if other
+   *   than `id` / `{@link Model#idAttribute idAttribute}`.
+   *
+   * @param {string=} otherKeyTarget
+   *
+   *   Column in the `Target` model's table which `otherKey` references, if other
+   *   than `Target` model's `id` / `{@link Model#idAttribute idAttribute}`.
+   *
    * @returns {Collection}
    */
-  belongsToMany(Target, joinTableName, foreignKey, otherKey) {
+  belongsToMany(Target, joinTableName, foreignKey, otherKey, foreignKeyTarget, otherKeyTarget) {
     return this._relation('belongsToMany', Target, {
-      joinTableName, foreignKey, otherKey
+      joinTableName, foreignKey, otherKey, foreignKeyTarget, otherKeyTarget
```

Comment:
In my [comment](https://github.com/tgriesser/bookshelf/pull/1397#issuecomment-250721474) I meant changing this to

```
{foreignKey, otherKey, targetIdAttribute: foreignKeyTarget, throughIdAttribute: otherKeyTarget}
```